From d4acc575f2d2e5cb0055820758ad109548e1dd36 Mon Sep 17 00:00:00 2001
From: Steve Kowalik <steven@wedontsleep.org>
Date: Fri, 19 Sep 2025 12:33:13 +1000
Subject: [PATCH] Remove use of SafeString

docutils 0.22.1 removed the SafeString class, since it isn't required
with Python 3 any more. Replace both uses of it with plain str.
---
 sphinx-jsonschema/__init__.py | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/sphinx-jsonschema/__init__.py b/sphinx-jsonschema/__init__.py
index b6875ea..ae46daa 100644
--- a/sphinx-jsonschema/__init__.py
+++ b/sphinx-jsonschema/__init__.py
@@ -27,7 +27,6 @@
 from docutils.parsers.rst import Directive, DirectiveError
 from docutils.parsers.rst import directives
 from docutils.utils import SystemMessagePropagation
-from docutils.utils.error_reporting import SafeString
 from .wide_format import WideFormat, NOESC
 
 
@@ -243,7 +242,7 @@ def get_json_data(self):
         except Exception as error:
             error = self.state_machine.reporter.error(
                 '"%s" directive encountered a the following error while parsing the data.\n %s'
-                % (self.name, SafeString("".join(format_exception_only(type(error), error)))),
+                % (self.name, str("".join(format_exception_only(type(error), error)))),
                 nodes.literal_block(schema, schema), line=self.lineno)
             raise SystemMessagePropagation(error)
 
@@ -253,7 +252,7 @@ def get_json_data(self):
             except KeyError:
                 error = self.state_machine.reporter.error(
                     '"%s" directive encountered a KeyError when trying to resolve the pointer'
-                    ' in schema: %s' % (self.name, SafeString(pointer)),
+                    ' in schema: %s' % (self.name, str(pointer)),
                     nodes.literal_block(schema, schema), line=self.lineno)
                 raise SystemMessagePropagation(error)
 
